src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2516,6 +2744,12 @@ ostree_sysroot_write_deployments_with_options (OstreeSysroot     *self,
   if (!_ostree_sysroot_ensure_writable (self, error))
     return FALSE;
 
+  /* for now, this is gated on an environment variable */
+  const char *opted_in = getenv ("OSTREE_EXP_AUTO_EARLY_PRUNE");
+  if (opted_in && *opted_in != '\0' && !opts->disable_auto_early_prune &&
+        !auto_early_prune_old_deployments (self, new_deployments, cancellable, error))
```

Comment:
We should also no-op this if there's not a distinct `/boot` filesystem right?

Reply:
Fixed!